### PR TITLE
few fixes in custom processing parameters

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
Fixes some issues in custom Processing parameters (based on the review comments in #39575)

- Make `QgsProcessingParameterVectorTileWriterLayers` public so it can be used in other algorithms.
- Mark `QgsProcessingParameterTypeTinInputLayers` and `QgsProcessingParameterTypeVectorTileWriterLayers` as private API
- Use member context in vector tile widget wrapper to avoid dangling pointers.

### CODE REVIEW

File: src/core/processing/qgsprocessingparametertininputlayers.h [hunk 1]
```diff
@@ -65,10 +65,13 @@ class CORE_EXPORT QgsProcessingParameterTinInputLayers: public QgsProcessingPara
     static QString typeName() { return QStringLiteral( "tininputlayers" ); }
 };
 
+///@cond PRIVATE
```
Comment (nyalldawson) [inline src/core/processing/qgsprocessingparametertininputlayers.h:68]: I'd sip skip this too -- it was never supposed to be public api, and there's no need for python code to be using it so no danger in removing it now.

File: src/core/processing/qgsprocessingparametervectortilewriterlayers.h [hunk 2]
```diff
@@ -66,6 +65,7 @@ class CORE_EXPORT QgsProcessingParameterVectorTileWriterLayers : public QgsProce
 
 };
 
+///@cond PRIVATE
```
Comment (nyalldawson) [inline src/core/processing/qgsprocessingparametervectortilewriterlayers.h:68]: Same here, sip skip this one too

**Comment by nyalldawson** [comment]
Thanks!

File: src/core/processing/qgsprocessingparametertininputlayers.h (hunk 1 above)
Comment (alexbruy) [reply src/core/processing/qgsprocessingparametertininputlayers.h:69]: Done.

File: src/core/processing/qgsprocessingparametervectortilewriterlayers.h (hunk 2 above)
Comment (alexbruy) [reply src/core/processing/qgsprocessingparametervectortilewriterlayers.h:69]: Done.

**Comment by alexbruy** [comment]
I also removed from SIP some other parameter types in addition to vector tiles and tin inputs.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**